weave/weave_client.py

```diff
@@ -832,14 +841,22 @@ def _save_nested_objects(self, obj: Any, name: Optional[str] = None) -> Any:
         # TODO: Kinda hacky way to dispatching Dataset with rows: Table
         elif isinstance(obj, WeaveObject) and hasattr(obj, "rows"):
             self._save_nested_objects(obj.rows)
+        elif get_serializer_for_obj(obj) is not None:
+            self._save_custom_obj(obj)
 
     @trace_sentry.global_trace_sentry.watch()
     def _save_table(self, table: Table) -> TableRef:
+        rows = table.rows
+
+        # TODO: Possible implementation of objects-in-cells support
```

Comment:
Leaving this here - need to sync with Andrew. This fixed my case, but broke mutations